### DISCUSSION TITLE
govmomi virtual machine linux

### DISCUSSION THREAD

**Discussion by feiqipengcheng** [vmware/govmomi#3630] [Community Help]
Do I want to obtain detailed partition information of a virtual machine (including the disk and partition name) through govmomi? If there is no corresponding interface, can Linux instructions be executed on the virtual machine through govmomi?

**Comment by dougm** [comment]
API provides [GuestDiskInfo](https://dp-downloads.broadcom.com/api-content/apis/API_VWSA_001/8.0U3/html/ReferenceGuides/vim.vm.GuestInfo.DiskInfo.html), but that doesn't have all partition details.

But you can use [GuestOperationsManager](https://dp-downloads.broadcom.com/api-content/apis/API_VWSA_001/8.0U3/html/ReferenceGuides/vim.vm.guest.GuestOperationsManager.html) to run commands in guest and transfer files in+out. There is a higher-level "toolbox" API around this in govmomi, see example:
https://github.com/vmware/govmomi/blob/46c15fb66eb121c238c15e9d17267e68f091a148/guest/toolbox/example_test.go#L48-L59
